### PR TITLE
refactor(textarea): remove legacy property and support for legacy syntax

### PR DESCRIPTION
**Pull request by brandyscarney**
Issue number: internal

---------

## What is the current behavior?

In Ionic Framework v7, we [simplified the textarea syntax](https://ionic.io/blog/ionic-7-is-here#simplified-form-control-syntax) so that it was no longer required to be placed inside of an `ion-item`. We maintained backwards compatibility by adding a `legacy` property which allowed it to continue to be styled properly when written in the following way:

```html
<ion-item>
  <ion-label>Label</ion-label>
  <ion-textarea></ion-textarea>
</ion-item>
```

While this was supported in v7, console warnings were logged to notify developers that they needed to update this syntax for the best accessibility experience.

## What is the new behavior?

- Removes the `legacy` property and support for the legacy syntax. Developers should follow the [migration guide](https://ionicframework.com/docs/api/textarea#migrating-from-legacy-textarea-syntax) in the textarea documentation to update their apps. The new syntax requires a `label` or `aria-label` on `ion-textarea`:
    ```html
    <ion-item>
      <ion-textarea label="Label"></ion-textarea>
    </ion-item>
    ```
- Removes the legacy tests under `textarea/test/legacy/` and all related screenshots
- Removes the textarea usage in `input/test/legacy/spec`, `item/test/disabled`, `item/test/legacy/disabled` and `item/test/legacy/fill`

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

1. Developers have had console warnings when using the legacy syntax since the v7 release. The migration guide for the new textarea syntax is outlined in the [Textarea documentation](https://ionicframework.com/docs/api/textarea#migrating-from-legacy-textarea-syntax).
2. This change has been documented in the Breaking Changes document with a link to the migration guide. 

BREAKING CHANGE:

The `legacy` property and support for the legacy syntax, which involved placing an `ion-textarea` inside of an `ion-item` with an `ion-label`, have been removed from textarea. For more information on migrating from the legacy textarea syntax, refer to the [Textarea documentation](https://ionicframework.com/docs/api/textarea#migrating-from-legacy-textarea-syntax).

### CODE REVIEW

File: core/src/components/item/test/disabled/item.e2e.ts-snapshots/item-disabled-diff-ios-ltr-Mobile-Chrome-linux.png
Comment (brandyscarney) [inline core/src/components/item/test/disabled/item.e2e.ts-snapshots/item-disabled-diff-ios-ltr-Mobile-Chrome-linux.png:1]: This screenshot is updated due to textarea being removed.

**Review by liamdebeasi — APPROVED** [review]

File: core/src/components/textarea/textarea.ios.scss
Comment (thetaPC) [inline core/src/components/textarea/textarea.ios.scss:1]: I'm still seeing an [instance](https://github.com/ionic-team/ionic-framework/blob/0c667612c6600e7ca31f33873cdea612210411c2/core/src/components/textarea/textarea.ios.scss#L31) of `.legacy-textarea` that needs to be removed.

File: core/src/components/textarea/textarea.md.scss
Comment (thetaPC) [inline core/src/components/textarea/textarea.md.scss:1]: I'm still seeing an [instance](https://github.com/ionic-team/ionic-framework/blob/0c667612c6600e7ca31f33873cdea612210411c2/core/src/components/textarea/textarea.md.scss#L52) of `.legacy-textarea` that needs to be removed.

**Review by thetaPC — CHANGES_REQUESTED** [review]

**Comment by thetaPC** [comment]
Shoot, I didn't realize it was merged in while I was reviewing. Apologies for being slow.

**Comment by brandyscarney** [comment]
I'm not sure how I missed those when I searched for `legacy` ha. Added a PR to remove them here: https://github.com/ionic-team/ionic-framework/pull/29006

You got added as a reviewer due to being a codeowner when I merged in this PR: https://github.com/ionic-team/ionic-framework/pull/28994 but they both had approvals so I merged this, sorry!

**Pull request merged**